### PR TITLE
Droth-1863_new_asset_roadworks

### PR DESCRIPTION
**Pull request by moraispedro**

### CODE REVIEW

File: digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/DamagedByThawFiller.scala [hunk 1]
```diff
@@ -32,18 +30,18 @@ class DamagedByThawFiller extends AssetFiller {
     }
 
     def toCurrentYear(period: DatePeriodValue): DatePeriodValue = {
-      val endDate = stringToDate(period.endDate)
-      val startDate = stringToDate(period.startDate)
+      val endDate = DateParser.stringToDate(period.endDate, DateParser.DatePropertyFormat)
```
Comment (oskarmalmstrom) [inline digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/DamagedByThawFiller.scala]: The stringToDate and dateToString methods used on the tests should be the same that is used here. The methods on this class can then also be removed.

File: digiroad2-oracle/src/main/resources/db/migration/V196__date_period_table_and_properties.sql [hunk 2]
```diff
@@ -7,19 +7,19 @@ create table date_period_value (
 );
 
 insert into localized_string (id, value_fi, value_sv, created_by, created_date)
-values (primary_key_seq.nextval, 'Kelirikkokausi', NULL, 'db_migration_v193', sysdate);
+values (primary_key_seq.nextval, 'Kelirikkokausi', NULL, 'db_migration_v196', sysdate);
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/resources/db/migration/V196__date_period_table_and_properties.sql]: flyway doesn't allow update, or revert the changes or create a new file V198

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/DamagedByThawService.scala [hunk 3]
```diff
@@ -7,24 +7,11 @@ import fi.liikennevirasto.digiroad2.service.RoadLinkService
...
-                val multiValue = value.asInstanceOf[DynamicValue]
-                //If exist at least one property to enrich with value all null properties value could be filter out
-                Some(multiValue.copy(value = DynamicAssetValue(multiValue.value.properties.filter(_.values.nonEmpty ) ++ props)))
-              case _ =>
-                Some(DynamicValue(DynamicAssetValue(props)))
-            }))
-            case _ => assets
-          }
-      }.toSeq
+      enrichWithProperties(properties, persistedLinearAsset)
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/DamagedByThawService.scala]: An actor will be needed for DamageByThaw

File: digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala [hunk 4]
```diff
@@ -91,6 +91,13 @@ class DynamicAssetUpdater(dynamicAssetService: DynamicLinearAssetService) extend
   }
 }
 
+class RoadWorksAssetUpdater(roadWorkAssetService: RoadWorkService) extends Actor {
```
Comment (oskarmalmstrom) [inline digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala]: Is a separate actor needed for this asset? The actor for dynamic linear assets also calls updateChangeSet

File: UI/src/assetTypeConfiguration.js [hunk 5]
```diff
@@ -42,7 +42,8 @@
       trWidthLimits: 370,
       manoeuvre: 380,
       careClass: 390,
-      carryingCapacity: 400
+      carryingCapacity: 400,
+      roadWorksAsset: 420
```
Comment (moraispedro) [inline UI/src/assetTypeConfiguration.js]: Is there any reason why roadWorkAsset was assigned with value 420 and not 410?

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala [hunk 6]
```diff
@@ -1460,7 +1464,7 @@ object DataFixture {
         }
       }
       additionalPanelIdToExpire.foreach { case (id, linkId, signType) =>
-//        trafficSignService.expireAssetWithoutTransaction(trafficSignService.withIds(Set(id).flatten), Some("batch_process_panel_merge"))
+        //        trafficSignService.expireAssetWithoutTransaction(trafficSignService.withIds(Set(id).flatten), Some("batch_process_panel_merge"))
```
Comment (moraispedro) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: Commented code

Comment (oskarmalmstrom) [reply digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: This is code is unrelated for this user story, but it should stay commented for now. The batch to merge traffic signs should not expire anything yet.

File: digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/DamagedByThawFiller.scala (hunk 1 above)
Comment (marquesfm) [reply digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/DamagedByThawFiller.scala]: Done

File: digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala (hunk 4 above)
Comment (marquesfm) [reply digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala]: Yes is needed, because on RoadWorkService we override the enrichPersistedLinearAssetProperties method and if we want to use him instead of enrichPersistedLinearAssetProperties of DynamicLinearAssetService we need to keep this.
